File: ssi-service/src/main/java/com/example/sns/SpanContextUtils.java
package com.example.sns;

import io.jaegertracing.internal.JaegerSpanContext;

import java.math.BigInteger;

public class SpanContextUtils {
    public static JaegerSpanContext getJaegerSpanContext(String uberTraceId) {
        String[] parts = uberTraceId.split(":");
        String traceIdString = parts[0];
        String spanIdString = parts[1];

        long traceId = new BigInteger(traceIdString, 16).longValue();
        long spanId = new BigInteger(spanIdString, 16).longValue();

        JaegerSpanContext spanContext = new JaegerSpanContext(0L, traceId, spanId, traceId, (byte) 1);
        return spanContext;
    }
}

File: solr-service/app/utils/SpanContextUtils.scala
package utils

import io.jaegertracing.internal.JaegerSpanContext

import java.math.BigInteger

object SpanContextUtils {
  def getSpanContext(uberTraceId: String) = {
    val parts: Array[String] = uberTraceId.split(":")
    val traceIdString: String = parts.headOption.getOrElse("")
    val spanIdString: String = parts.lift(1).getOrElse("")

    val traceId = new BigInteger(traceIdString, 16).longValue()
    val spanId = new BigInteger(spanIdString, 16).longValue()

    val spanContext = new JaegerSpanContext(0, traceId, spanId, traceId, 1);
    spanContext
  }
}
